lib/puppet_x/puppetlabs/kubernetes/provider.rb

```diff
@@ -208,13 +208,23 @@ def self.method_missing(method_sym, *arguments, &block)
         end
 
         def destroy_instance_of(type, name)
-          call("delete_#{type}", name, namespace)
+          call("delete_#{type}", kubename, namespace)
         end
 
         def call(method, *object)
           self.class.call(method, *object)
         end
 
+        def kubename
+          # In any case, we need to provide a name.So we either use the one from puppet
+          # or the one from the kubernetes api
+          if self.metadata == :absent
+            resource[:metadata]['name']
```

Comment:
There's no guarantee there's a `:metadata` in the current puppet `resource` (like for `kubernetes_namespace`).
Maybe do what has been done for the namespace:
~~~ruby
            if resource[:metadata]
              resource[:metadata]['name']
            else
              # for resources without metadata like namespaces
              # name will be given by the Puppet resource
              nil
            end
~~~

Reply:
The namespace name is set into metadata/name. I think every kubernetes resource must have a metadata/name